### PR TITLE
Add patch for software echo cancellation under Windows

### PR DESCRIPTION
**Pull request by logist322**
## Synopsis

This PR adds the patch to exclude hardware AEC under Windows.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains `Draft: ` prefix
    - [x] Name contains issue reference
    - [x] Has `k::` labels applied
    - [x] Has assignee
- [x] Documentation is updated (if required)
- [x] Tests are updated (if required)
- [x] Changes conform code style
- [x] CHANGELOG entry is added (if required)
- [x] FCM (final commit message) is posted
    - [x] and approved
- [x] [Review][l:2] is completed and changes are approved
- Before merge:
    - [ ] Milestone is set
    - [ ] PR's name and description are correct and up-to-date
    - [ ] `Draft: ` prefix is removed
    - [ ] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by logist322** [comment]
## FCM
`Add the patch to exclude hardware AEC under Windows (#3)`

**Review by alexlapa — APPROVED** [review]

**Review by tyranron — APPROVED** [review]

File: patch/webrtc_voice_engine.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+--- a/src/media/engine/webrtc_voice_engine.cc
++++ b/src/media/engine/webrtc_voice_engine.cc
+@@ -477,6 +477,7 @@ bool WebRtcVoiceEngine::ApplyOptions(const AudioOptions& options_in) {
+   }
+ #endif
+ 
++#if !defined(WEBRTC_WIN)
+   if (options.echo_cancellation) {
```
[the remark this answers is not in the document]
Comment (tyranron) [reply patch/webrtc_voice_engine.patch:8]: @logist322 @alexlapa ping

[the remark this answers is not in the document]
Comment (logist322) [reply patch/webrtc_voice_engine.patch:8]: @tyranron do you mean just cut this piece out for Windows?

[the remark this answers is not in the document]
Comment (tyranron) [reply patch/webrtc_voice_engine.patch:8]: @logist322 no, I mean to specify `options.echo_cancellation = false`, so this block won't fire. Without customizing build. Is it possible?

[the remark this answers is not in the document]
Comment (tyranron) [reply patch/webrtc_voice_engine.patch:8]: @logist322 

> no, it is not. `options.echo_cancellation` is used later in the same method to check whether we need to enable software echo-cancellation (AEC3). `options.echo_cancellation` is specified in the `cricket::AudioOptions` when we are creating an `audio source` in our `libwebrtc-sys` crate.

Ok, then this solution is good enough. Let's go with it.

**Pull request merged**